pkg/volume/plugins.go

```diff
@@ -666,7 +666,7 @@ func (pm *VolumePluginMgr) FindAttachablePluginBySpec(spec *Spec) (AttachableVol
 	if attachableVolumePlugin, ok := volumePlugin.(AttachableVolumePlugin); ok {
 		return attachableVolumePlugin, nil
 	}
-	return nil, nil
+	return nil, fmt.Errorf("no attachable volume plugin matched")
```

Comment:
Should we return an error in here? I think we should only return an error if we can't find the volume plugin. If we can find the plugin, but the plugin is not attachable(expandable), we just return an nil error with a nil plugin, and the caller should handle the nil plugin properly.

Reply:
Agree

Reply:
I find some other places where we call `Find...PluginBy...`,  and we just check if err is returned, if not, call volume plugin functions directly . So it is hard for us to ensure that everyone will check if volume plugin is nil, so returning err here will help us, just like we do in many other functions . e.g.
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/plugins.go#L576
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/plugins.go#L602.

Reply:
Yes, I can see the trouble. But from code wise, I agree with @mlmhl, I just submitted a similar PR at #57677. Anyway, return error can also fix the problem

Reply:
hrmm - I think the error was in code that was calling `FindExpandablePluginBySpec` and not checking if returned value is nil. I am not sure if we want to return an error from all these code paths as well, because a `nil` return value implicitly means we couldn't find the plugin and hence it is equivalent to returning `fmt.Errorf("no attachable volume plugin found")`.

I think we should just fix the code that calls `FindExpandablePluginBySpec` and make sure it also checks for volume plugin's nil possibility before proceeding with resizing.

Reply:
I can change back to the original value `nil` although i still prefer returning err.